base/range.jl

```diff
@@ -32,8 +32,8 @@ _colon(::Any, ::Any, start::T, step, stop::T) where {T} =
     (:)(start, [step], stop)
 
 Range operator. `a:b` constructs a range from `a` to `b` with a step size
-of 1 (often a [`UnitRange`](@ref)), and `a:s:b` is similar but uses a step
-size of `s` (a [`StepRange`](@ref) or [`StepRangeLen`](@ref)).
+equal to 1 (a [`UnitRange`](@ref) or [`StepRangeLen`](@ref)), and `a:s:b` is
```

Comment:
There are cases where `a:b` returns neither a `UnitRange` nor a `StepRangeLen`, even in `Base`:
```
julia> typeof('a':'d')
StepRange{Char, Int64}
```
Furthermore, anyone can define their own types and `:` methods. For example, [InfiniteArrays.jl](https://github.com/JuliaArrays/InfiniteArrays.jl) defines `1:∞` to return an `InfUnitRange{Int}` (a type that is also defined in that package).

Reply:
I agree that there are cases where `a:b` returns neither a `UnitRange` nor a `StepRangeLen`, even in `Base`, as you've pointed out, but adding the behaviour of what outside packages would do is not "useful" here, since anyone can extend it to what they want.

The aim of docstrings is to provide information on what is the expected behaviour of a function in `Base`; so should we extend that docstring to add `StepRange`, or just leave it all out?

Reply:
I think the updated docstrings serves all parties.